File: lib/python/openlinktoken/src/test/openlinktoken/attributes/attribute_field_test.py
# SPDX-License-Identifier: MIT

import pytest

from openlinktoken.attributes.attribute_field import AttributeField
from openlinktoken.attributes.general.string_attribute import StringAttribute
from openlinktoken.attributes.person.first_name_attribute import FirstNameAttribute
from openlinktoken.attributes.person.last_name_attribute import LastNameAttribute


class TestAttributeField:
    """Test cases for AttributeField."""

    def test_constructor(self):
        """Test basic field construction."""
        field = AttributeField("LastName", LastNameAttribute)
        assert field.field_id == "LastName"
        assert field.attribute_class == LastNameAttribute

    def test_constructor_rejects_none_field_id(self):
        """Test that None field_id raises ValueError."""
        with pytest.raises(ValueError, match="field_id must not be None"):
            AttributeField(None, StringAttribute)

    def test_constructor_rejects_none_attribute_class(self):
        """Test that None attribute_class raises ValueError."""
        with pytest.raises(ValueError, match="attribute_class must not be None"):
            AttributeField("Test", None)

    def test_equality_by_field_id(self):
        """Test that equality is determined by field_id only."""
        field1 = AttributeField("Name", StringAttribute)
        field2 = AttributeField("Name", FirstNameAttribute)
        assert field1 == field2
        assert hash(field1) == hash(field2)

    def test_inequality_by_field_id(self):
        """Test that different field_ids produce inequality."""
        field1 = AttributeField("FirstName", StringAttribute)
        field2 = AttributeField("LastName", StringAttribute)
        assert field1 != field2

    def test_repr(self):
        """Test string representation."""
        field = AttributeField("BirthDate", StringAttribute)
        assert "BirthDate" in repr(field)
        assert "StringAttribute" in repr(field)

File: lib/java/openlinktoken/src/test/java/org/openlinktoken/attributes/AttributeFieldTest.java
/* SPDX-License-Identifier: MIT */
package org.openlinktoken.attributes;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

import org.junit.jupiter.api.Test;

import org.openlinktoken.attributes.general.StringAttribute;
import org.openlinktoken.attributes.person.FirstNameAttribute;
import org.openlinktoken.attributes.person.LastNameAttribute;

class AttributeFieldTest {

    @Test
    void testConstructor() {
        var field = new AttributeField("LastName", LastNameAttribute.class);
        assertEquals("LastName", field.getFieldId());
        assertEquals(LastNameAttribute.class, field.getAttributeClass());
    }

    @Test
    void testConstructorRejectsNullFieldId() {
        assertThrows(NullPointerException.class, () -> new AttributeField(null, StringAttribute.class));
    }

    @Test
    void testConstructorRejectsNullAttributeClass() {
        assertThrows(NullPointerException.class, () -> new AttributeField("Test", null));
    }

    @Test
    void testEqualityByFieldId() {
        var field1 = new AttributeField("Name", StringAttribute.class);
        var field2 = new AttributeField("Name", FirstNameAttribute.class);
        assertEquals(field1, field2);
        assertEquals(field1.hashCode(), field2.hashCode());
    }

    @Test
    void testInequalityByFieldId() {
        var field1 = new AttributeField("FirstName", StringAttribute.class);
        var field2 = new AttributeField("LastName", StringAttribute.class);
        assertNotEquals(field1, field2);
    }

    @Test
    void testToString() {
        var field = new AttributeField("BirthDate", StringAttribute.class);
        var str = field.toString();
        assertEquals("AttributeField{fieldId='BirthDate', attributeClass=StringAttribute}", str);
    }
}
